src/signal.rs

```diff
@@ -183,21 +159,30 @@ pub fn validate_signal_num(num: c_int, for_vcpu: bool) -> errno::Result<c_int> {
 /// use vmm_sys_util::signal::{register_signal_handler, SignalHandler};
 ///
 /// extern "C" fn handle_signal(_: c_int, _: *mut siginfo_t, _: *mut c_void) {}
-/// unsafe {
-///     register_signal_handler(0, SignalHandler::Siginfo(handle_signal), true, SA_SIGINFO);
-/// }
+/// register_signal_handler(0, handle_signal, true, SA_SIGINFO);
 /// ```
 
-pub unsafe fn register_signal_handler(
+pub fn register_signal_handler(
     num: c_int,
     handler: SignalHandler,
     for_vcpu: bool,
     flag: c_int,
 ) -> errno::Result<()> {
     let num = validate_signal_num(num, for_vcpu)?;
-    let mut act: sigaction = handler.into();
-    SignalHandler::set_flags(&mut act, flag);
-    match sigaction(num, &act, null_mut()) {
+
+    // Safe, because this is a POD struct.
+    let mut act: sigaction = unsafe { mem::zeroed() };
+    act.sa_sigaction = handler as *const () as usize;
+    act.sa_flags = flag;
```

Comment:
When using `sa_sigaction` instead of `sa_handler` to specify the handler, `sa_flags` needs to have `libc::SA_SIGINFO` set.

So at a minimum this line should be:
```rust
act.sa_flags = flag | libc::SA_SIGINFO
```

Looking at the other possible values of `sa_flags` I suggest we simply don't expose this as none of them seem like something VM's should use.

My preference is removing `flag` altogether from this function's param list and just set `sa_flags = SA_SIGINFO`.